Feature/rest integration

maciejBart99 opened the pull request.

maciejBart99 committed:
refactor(todo): Add abstraction layer to todo data operations

Add abstract class for todos data access

maciejBart99 committed:
feat(todo): Add rest api data source for todos

Add service to manage http calls to todo's rest api and proxy configuration for angular

MaciejSikorski commented on libs/todo/data-access/src/lib/services/todo-rest.service.ts:
```diff
@@ -0,0 +1,29 @@
+export class TodoRestService extends TodoDataService {
+
+  private readonly resourceUrl = 'todos';
+  private readonly headers = {'Content-Type': 'application/json'};
```
Isn't it included by default?

MaciejSikorski commented on libs/todo/data-access/src/lib/services/todo-rest.service.ts:
```diff
@@ -0,0 +1,29 @@
+      return this.http.post<any>(this.resourceUrl, newTodo, {headers: this.headers});
+  }
+  editTodo(entity: TodosEntity, patch: Partial<TodosEntity>): Observable<any> {
+    return this.http.patch<any>(`${this.resourceUrl}/${entity.id}`, patch, {headers: this.headers});
```
try to avoid any

MaciejSikorski commented on libs/todo/data-access/src/lib/services/todo-local-storage.service.ts:
```diff
@@ -2,11 +2,10 @@ import { Injectable } from '@angular/core';
-})
-export class TodoLocalStorageService {
+@Injectable()
+export class TodoLocalStorageService extends TodoDataService {
```
use implements

maciejBart99 committed:
refactor(todo): Change data source class inheritance

Change inheritance into interface implementation in todo data source classes

maciejBart99 committed:
refactor(todo): Remove unnecesary headers and request response typing

Remove explicit application/json header from todo rest api service
Add typings to todod rest api service
